src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -203,6 +211,26 @@ class Peer(
                 input.send(WrappedChannelEvent(it.channelId, ChannelEvent.WatchReceived(it)))
             }
         }
+        launch {
+            finalWallet.walletStateFlow.collect { wallet ->
+                if (wallet.spendableBalance > 0.sat) {
+                    logger.info { "${wallet.spendableBalance} available on final wallet" }
+                }
+            }
+        }
+        launch {
+            swapInWallet.walletStateFlow
+                .takeWhile { wallet ->
+                    if (wallet.spendableBalance > 10_000.sat) {
+                        logger.info { "${wallet.spendableBalance} available on swap-in wallet, requesting channel" }
+                        input.send(RequestChannelOpen(randomBytes32(), wallet, maxFeesBasisPoint = 100))
```

Comment:
Note that we mustn't forget to fetch `maxFeesBasisPoint` from the user's settings somehow.